Firebase/Auth/Source/FIRAuth.m

```diff
@@ -438,7 +439,21 @@ - (instancetype)initWithAPIKey:(NSString *)APIKey appName:(NSString *)appName {
     _settings = [[FIRAuthSettings alloc] init];
     _firebaseAppName = [appName copy];
     #if TARGET_OS_IOS
-    UIApplication *application = [UIApplication sharedApplication];
+
+    static Class applicationClass = nil;
+    // iOS App extensions should not call [UIApplication sharedApplication], even
+    // if UIApplication responds to it.
+    if (![FIRAppEnvironmentUtil isAppExtension]) {
+      Class cls = NSClassFromString(@"UIApplication");
+      if (cls && [cls respondsToSelector:NSSelectorFromString(@"sharedApplication")]) {
+        applicationClass = cls;
+      }
+    }
+    UIApplication *application = nil;
+    if (applicationClass) {
```

Comment:
If is redundant since calls messages to nil are ignored.

Reply:
Done.